src/network/networkd-route.c

```diff
@@ -1231,6 +1231,50 @@ int config_parse_route_mtu(
         return 0;
 }
 
+int config_parse_route_device(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Network *network = userdata;
+        _cleanup_(route_free_or_set_invalidp) Route *n = NULL;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = route_new_static(network, filename, section_line, &n);
+        if (r < 0)
+                return r;
+
+        r = config_parse_bool(unit, filename, line, section, section_line, lvalue, ltype, rvalue, &n->on_device, userdata);
+        if (r < 0)
+                return r;
+
+        /* Default gateway on device works only on IPv4 */
+        r = inet_pton(AF_INET, "0.0.0.0", &n->gw);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Failed to parse gateway address '0.0.0.0', ignoring assignment: %m");
+                return 0;
+        }
+
+        n->family = AF_INET;
```

Comment:
Maybe, before setting the value, please check that `n->family` is not set or `AF_INET`, that is please make sure that the setting is not combined with any IPv6 settings.